g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java

```diff
@@ -91,9 +97,159 @@ public APIResponseDTO getMultTrans(String productName, String version,String pse
+        }else {
+            return getV2MultipleComponentsTrans(productName, componentsStr,  version,  localesStr,  pseudo, req);
+        }
+        
+        
+    }
+   
+    
+     private TranslationDTO getPartTranslationReqParams(String productName,
+             String components, String version, String locales, String pseudo,
+             HttpServletRequest req) throws Exception {
+         TranslationDTO translationDTO = new TranslationDTO();
+         translationDTO.setProductName(productName);
+         translationDTO.setVersion(version);
+         List<String> componentList = null;
+         if (StringUtils.isEmpty(components)) {
+             componentList = productService.getComponentNameList(productName, version);
+         }else {
+             componentList = new ArrayList<String>();
+             for (String component : components.split(",")) {
+                 componentList.add(component.trim());
+             }
+         }
+         translationDTO.setComponents(componentList);
+         List<String> localeList = new ArrayList<String>();
+         if (new Boolean(pseudo)) {
+             localeList.add(ConstantsKeys.LATEST);
+         } else if (!StringUtils.isEmpty(locales)) {
+             for (String locale : locales.split(",")) {
+                 localeList.add(getMapingLocale(productName, version, locale.trim()));
+             }      
+         } else {
+             localeList = productService.getSupportedLocaleList(productName,version);
+         }
+         translationDTO.setLocales(localeList);
+         translationDTO.setPseudo(new Boolean(pseudo));
+         return translationDTO;
+         
+     }
+    
+     
+     
+     
+    
+     
+    
+     @SuppressWarnings("unchecked")
+    private JSONObject getNUllBundle(String component, String locale) {
+         JSONObject object = new JSONObject();
+         object.put("locale", locale);
+         object.put("component", component);
+         object.put("messages", null);
+         return object;
+         
+     }
+     
+     
+     private JSONObject getBundle(String component, String locale, TranslationDTO allTranslationDTO) {
+         
+         JSONArray array = allTranslationDTO.getBundles();
+         @SuppressWarnings("unchecked")
+        Iterator<JSONObject> objectIterator =  array.iterator();
+         
+         while(objectIterator.hasNext()) {
+             JSONObject object = objectIterator.next();
+             String fileLocale = (String) object.get(ConstantsKeys.lOCALE);
+             String fileComponent = (String) object.get(ConstantsKeys.COMPONENT);
+             if(locale.equals(fileLocale)&& component.equals(fileComponent)) {    
+                 return object;
+             }
+         }
+         
+        return null; 
+     }
 
+     
+    /**
+     *  get the API v2 mult-component translation
+     */
+      @SuppressWarnings("unchecked")
+    public APIResponseDTO getV2MultipleComponentsTrans(String productName,
```

Comment:
change the name 'getV2MultipleComponentsTrans' to 'getPartialComTrans'

Reply:
done